### DISCUSSION TITLE
Récupération clé d'acteur en Javascript pour renvois dans additionnelles d'une modale

### DISCUSSION THREAD

**Discussion by Synfactory** [witsa/synapps#722] [Q&A] [answered]
Bonjour,
est-il possible de récupérer la clé d'un acteur en Javascript?
![image](https://github.com/witsa/synapps/assets/165296029/0836535a-936f-42f9-9029-efe7804787ac)
Je pense que c'est possible mais je ne connais pas trop la syntaxe.

Merci de vos retours :)

**Comment by Synfactory** [comment, marked as answer]
J'ai réussi à "bidouiller" pour que ca fonctionne avec des additionnelles !
![image](https://github.com/witsa/synapps/assets/165296029/4af485f1-6d1e-4355-8c1f-ed9b41e8aa38)

**Comment by manufitoussiwit** [comment]
Salut, pour récupérer la clé d'un acteur, c'est tout simplement : `.key`.

Il y a une doc de l'API Javascript (qui est en contruction encore...), on peut y voir l'accès à la clé d'un acteur par exemple sur un composite : https://witsa.github.io/synapps/script-api/Actor.Composite.html#key

**Reply by manufitoussiwit** [reply]
Dans votre `onClick` : 
```js
// this est l'acteur composite ici.
context.getActor('modal1').properties.propCheminVar = this.key;
```

**Reply by Synfactory** [reply]
Hello,
merci pour le retour. C'est bien compris.
